src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyValidateController.java

```diff
@@ -117,14 +134,32 @@ private boolean checkKeyAgainstVaultConfig(byte[] key) {
 	}
 
 	private void validateRecoveryKey() {
-		isWrongKey = false;
-		var valid = recoveryKeyFactory.validateRecoveryKey(recoveryKey.get(), unverifiedVaultConfig != null ? this::checkKeyAgainstVaultConfig : null);
-		if (valid) {
-			recoveryKeyState.set(RecoveryKeyState.CORRECT);
-		} else if (isWrongKey) { //set via side effect in checkKeyAgainstVaultConfig()
-			recoveryKeyState.set(RecoveryKeyState.WRONG);
-		} else {
-			recoveryKeyState.set(RecoveryKeyState.INVALID);
+		switch (recoverType.get()) {
+			case RESTORE_VAULT_CONFIG -> {
+				AtomicBoolean illegalArgumentExceptionOccurred = new AtomicBoolean(false);
+				var combo = RecoverUtil.validateRecoveryKeyAndGetCombo(
```

Comment:
Also here, the exception should be catched with try-catch, not with a atomic boolean, which is handed over the the method.